client/my-sites/stats/site.jsx

```diff
@@ -261,613 +313,501 @@ class StatsSite extends Component {
 		if ( ! paramsValid ) {
 			return undefined;
 		}
 
-		let url = `/stats/${ period.period }/${ path }/${ slug }`;
+		let url = `/stats/${ props.period.period }/${ modulePath }/${ slug }`;
 
 		if ( query?.start_date ) {
 			url += `?startDate=${ query.start_date }&endDate=${ query.date }`;
 		} else {
-			url += `?startDate=${ period.endOf.format( DATE_FORMAT ) }`;
+			url += `?startDate=${ props.period.endOf.format( DATE_FORMAT ) }`;
 		}
 
 		return url;
-	}
+	};
 
-	renderStats( isInternal ) {
-		const {
-			date,
-			siteId,
-			slug,
-			isAtomic,
-			isJetpack,
-			isSitePrivate,
-			isOdysseyStats,
-			context,
-			supportsPlanUsage,
-			supportsUTMStatsFeature,
-			supportsDevicesStatsFeature,
-			isOldJetpack,
-			hasSiteLoadedFeatures,
-			shouldForceDefaultDateRange,
-			shouldForceDefaultPeriod,
-			supportUserFeedback,
-			momentSiteZone,
-			wpcomShowUpsell,
-		} = this.props;
-		const shouldShowUpsells = isOdysseyStats && ! isAtomic;
-		const supportsUTMStats = supportsUTMStatsFeature || isInternal;
-		const supportsDevicesStats = supportsDevicesStatsFeature || isInternal;
-
-		const queryDate = date.format( DATE_FORMAT );
-		const { period } = this.props.period;
-		const moduleStrings = statsStrings();
-
-		// Set up a custom range for the chart.
-		// Dependant on new date range picker controls.
-		let customChartRange = null;
-
-		// Sort out end date for chart.
-		const chartEnd = this.getValidDateOrNullFromInput( context.query?.chartEnd, 'endDate' );
-
-		if ( chartEnd ) {
-			customChartRange = { chartEnd };
-		} else {
-			customChartRange = {
-				chartEnd: momentSiteZone.format( DATE_FORMAT ),
-			};
+	useMemo( () => {
+		const activeTab = getActiveTab( chartTab );
+		if ( activeTab !== activeTabState ) {
+			setActiveTabState( activeTab );
+			setActiveLegend( period !== 'hour' ? activeTab.legendOptions || [] : [] );
 		}
+	}, [ chartTab, period, activeTabState ] );
 
-		// Find the quantity of bars for the chart.
-		let daysInRange = this.getDefaultDaysForPeriod( period );
-		const chartStart = this.getValidDateOrNullFromInput( context.query?.chartStart, 'startDate' );
-		const isSameOrBefore = moment( chartStart ).isSameOrBefore( moment( chartEnd ) );
-
-		if ( chartStart && isSameOrBefore ) {
-			// Add one to calculation to include the start date.
-			daysInRange = moment( chartEnd ).diff( moment( chartStart ), 'days' ) + 1;
-			customChartRange.chartStart = chartStart;
-		} else {
-			// if start date is missing let the frequency of data take over to avoid showing one bar
-			// (e.g. months defaulting to 30 days and showing one point)
-			customChartRange.chartStart = momentSiteZone
-				.clone()
-				.subtract( daysInRange - 1, 'days' )
-				.format( DATE_FORMAT );
-		}
+	// Sort out end date for chart.
+	const chartEnd = getValidDateOrNullFromInput( context.query?.chartEnd, 'endDate' );
 
-		customChartRange.daysInRange = daysInRange;
+	if ( chartEnd ) {
```

Comment:
```suggestion
	let customChartRange = null;

	if ( chartEnd ) {
```
The [variable](https://github.com/Automattic/wp-calypso/pull/97838/files#diff-ea5f237b7989be993ecfab2290e70ff69102850504ef05474ce36ff414a5afacR216) `customChartRange` is more readable if declared here.

Reply:
I agree. Addressed in bf3e5cd45a2